src/classes/CRLP_Rollup_SEL.cls

```diff
@@ -178,43 +444,65 @@ public class CRLP_Rollup_SEL {
-        if (UserInfo.isMultiCurrencyOrganization() && allFields.containsKey('currencyisocode')) {
-            referencedFields.put('CurrencyIsoCode', allFields.get('currencyisocode'));
+
+        if (objType != null) {
+            final DescribeSObjectResult sObjectDescribe = UTIL_Describe.getObjectDescribe(objType);
+            final String sObjectName = sObjectDescribe.getName();
+            
+            // Note: DescribeSObjectResult.fields.getMap() is a case-insensitive map
+            final Map<String, SObjectField> fieldsByName = sObjectDescribe.fields.getMap();
+
+            // Always include the CurrencyIsoCode field if available for the object
+            if (
+                UserInfo.isMultiCurrencyOrganization() 
+                && fieldsByName.get('CurrencyIsoCode') != null
+            ) {
+                fields.add(fieldsByName.get('CurrencyIsoCode'));
+            }
+    
+            // Summary Field and Rollup Field from cachedRollups
+            {   
+                final RollupField[] rollupFields = getRollupFields();
+
+                for (Rollup__mdt rollup : cachedRollups) {
+                    if (rollup.Active__c == true) {
+                        // Summary Field
+                        if (rollup.Summary_Object__r.QualifiedApiName == sObjectName) {
+                            final SObjectField field = fieldsByName.get(rollup.Summary_Field__r.QualifiedApiName);
+                            if (field != null) {
+                                fields.add(field);
+                            }
+                        }
+
+                        // Rollup Fields
+                        for (RollupField rollupField : rollupFields) {
+                            if (rollupField.isFieldUsed(rollup, sObjectName)) {
+                                final SObjectField field = fieldsByName.get(rollupField.getFieldName(rollup));
+                                if (field != null) {
+                                    fields.add(field);
+                                }
+                            }
+                        }
+                    }
+                }
+            }   
         }
-        // Always add the AccountId and PrimaryContactId for the Opportunity object
-        if (objType == Opportunity.SObjectType) {
-            String fldName = SObjectType.Opportunity.fields.AccountId.getName();
-            referencedFields.put(fldName, allFields.get(fldName.toLowerCase()));
-            fldName = SObjectType.Opportunity.fields.Primary_Contact__c.getName();
-            referencedFields.put(fldName, allFields.get(fldName.toLowerCase()));
+
+        final List<Schema.DescribeFieldResult> referencedFields = new List<Schema.DescribeFieldResult>();
+        // Add fields' Describes to referencedFields
+        for (SObjectField field : fields) {
+            referencedFields.add(field.getDescribe());
         }
 
-        validateReferencedFields(referencedFields, objName);
```

Comment:
Old Way
------
1. Get a Map: Field Name --> `DescribeFieldResult` for `objType`.
2. Put values without null-checking to return in a Map from Map in Step 1.
3. Since we didn't null check in step 2, call `validatedReferencedFields` which throws a `CRLP_Rollup_SVC.CRLP_Exception` if any values from Step 2 turn out to be null.

New Way
----------
1. Store `objType.getDescribe().fields.getMap()`
2. Collect fields as a `Set<SObjectField>` and don't add non-mappable values
3.  Loop through `SObjectFields` and collect and return their `DescribeFieldResults`

Reply:
Per conversation with @force2b , we don't need a validate method since `SObjectFields` are collected and `field.getDescribe()` is called for each `SObjectField`.  If a `SObjectField` is null, a NullPointerException will be thrown which is what `validateReferencedFields(...)` does.

@force2b  says the Exception type doesn't matter as long as an Exception is thrown.